tests/TestCase/ORM/TableTest.php

```diff
@@ -3070,6 +3071,48 @@ public function testSaveManyOrFailFailed(): void
         $table->saveManyOrFail($entities);
     }
 
+    public function testSaveWithBuildRulesFailWithErrorMessage(): void
+    {
+        $Articles = new class extends Table {
+            public function initialize(array $config): void
+            {
+                $this->setAlias('Articles');
+                $this->setTable('articles');
+                $this->hasMany('Comments');
+            }
+        };
+        $Comments = new class extends Table {
+            public function initialize(array $config): void
+            {
+                $this->setAlias('Comments');
+                $this->setTable('comments');
+            }
+
+            public function buildRules(RulesChecker $rules): RulesChecker
+            {
+                return $rules->add(function () {
+                    return 'Xyz';
+                });
+            }
+        };
+        TableRegistry::getTableLocator()->set('Comments', $Comments);
+
+        $article = $Articles->newEntity([
+            'title' => 'First Article',
+            'body' => 'First Article Body',
+            'published' => 'Y',
+            'comments' => [
+                '_ids' => [1],
+            ],
+        ]);
+
+        $result = $Articles->save($article, ['associated' => ['Comments']]);
+        $this->assertFalse($result);
+
+        // There should be errors here, due to comment not being savable.
+        debug($article->getErrors());
```

Comment:
Yeah this doesn't look right to me. If the save failed there should be an error available. Without that the developer and end user have no idea what is wrong.